src/XR/features/WebXRLightEstimation.ts

```diff
@@ -0,0 +1,161 @@
+     */
+    public onReflectionCubeMapUpdatedObservable: Observable<WebGLTexture> = new Observable();
+
+    /**
+    * Creates a new instance of the light estimation feature
+    * @param _xrSessionManager an instance of WebXRSessionManager
+    * @param options options to use when constructing this feature
+    */
+    constructor(
+        _xrSessionManager: WebXRSessionManager,
+        /**
+         * options to use when constructing this feature
+         */
+        public readonly options: IWebXRLightEstimationOptions
+    ) {
+        super(_xrSessionManager);
+        this.xrNativeFeatureName = "light-estimation";
+
+        // https://immersive-web.github.io/lighting-estimation/
+        Tools.Warn("light-estimation is an experimental and unstable feature.");
+    }
+
+    /**
+     * While the estimated cube map is expected to update over time to better reflect the user's environment as they move around those changes are unlikely to happen with every XRFrame.
+     * Since creating and processing the cube map is potentially expensive, especially if mip maps are needed, you can listen to the onReflectionCubeMapUpdatedObservable to determine
+     * when it has been updated.
+     */
+    public get reflectionCubeMap(): Nullable<WebGLTexture> {
+        return this._reflectionCubeMap;
+    }
+
+    /**
+     * The XRLightProbe object created during attach (may not be available for a few frames depending on device).
+     *
+     * The XRLightProbe itself contains no lighting values, but is used to retrieve the current lighting state with each XRFrame.
+     * XR Light Probe also contains the probe space, may update it's pose over time as the user moves around their environment.
+     */
+    public get xrLightProbe(): Nullable<XRLightProbe> {
+        return this._xrLightProbe;
+    }
+
+    /**
+     * The most recent light estimate.  Available starting on the first frame where the device provides a light probe.
+     */
+    public get xrLightingEstimate(): Nullable<XRLightEstimate> {
+        return this._xrLightEstimate;
+    }
+
+    private _getXRGLBinding(): XRWebGLBinding {
+        if (this._xrWebGLBinding === null) {
+            // TODO: find a way that looks less hacky to get webgl context.  try webgl2 and fallback to webgl.
+            const canvas: HTMLCanvasElement = this._xrSessionManager.scene.getEngine().getRenderingCanvas()!;
+            let context: Nullable<WebGLRenderingContext | WebGL2RenderingContext> = canvas.getContext("webgl2");
+            if (!context) {
+                context = canvas.getContext("webgl")!;
+            }
+            this._xrWebGLBinding = new XRWebGLBinding(this._xrSessionManager.session, context);
+        }
+        return this._xrWebGLBinding;
+    }
+
+    /**
+     * Event Listener to for "reflectionchange" events.
+     */
+    private _updateReflectionCubeMap = (): void => {
+        this._reflectionCubeMap = this._getXRGLBinding().getReflectionCubeMap(this._xrLightProbe!);

```

Comment:
does the old one needs to be disposed?

Reply:
@RaananW  I didn't clean up based on this:
https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.8
```
Note that underlying GL object will be automatically marked for deletion when the JS object is destroyed, however this method allows authors to mark an object for deletion early.
```

It's easy for me to mark for deletion  - I started making some changes to do so.  Shall I update with `context.deleteTexture(this._reflectionCubeMap)`?